### DISCUSSION TITLE
readTimeout Clarification

### DISCUSSION THREAD

**Discussion by AdiAbuAli** [bluenviron/mediamtx#799] [Questions]
Hi,
Am using wyzecam, and mrlt8_Wyze_bridge, and Synology as NVR.
For readTimeout parameter (defualt is 10s), is this timer between the camera and RTSP_SIMPLE_SERVER,  or between RTSP_SIMPLE_SERVER and the NVR?

And what is the difference between this and KeepAlive between NVR and RTSP_SIMPLE_SERVER (if this timer for that interface)?

How this timer work? 
And lastly, Writetimeout is just the opposite?

P.S.: both mrlt8 docker, simple RTSP server, NVR, are in same platform (Synology).

Thanks,

**Comment by aler9** [comment]
Hello, `readTimeout` is the maximum response time between a server request and the camera response. After this, the camera is disconnected from the server.

The keepalive mechanism doesn't use readTimeout, but uses an interval that is either set by the camera or 30secs by default.

`writeTimeout` is the maximum write time of every server request.

**Comment by AdiAbuAli** [comment]
Thank you for the clarification....

Thanks,
________________________________

**Comment by AdiAbuAli** [comment]
But can you clarify more on "writeTimeout is the maximum write time of every server request."
What type or examples server of requests an to which interface, to camera or NVR?
________________________________

**Reply by aler9** [reply]
WriteTimeout is applied to all TCP writes from the server to any external client. If you have any more doubts you should check the source code.

[a comment by an automated account is not shown]
